### ISSUE TITLE
publish second sector size (1GiB)

### ISSUE THREAD

**Issue by porcuquine** [filecoin-project/rust-fil-proofs#625]
### Description

NOTE: this work should be sequenced before #626.

We should begin to support larger sector sizes by adding a second sector size while continuing to support the existing (currently 256MiB) sector size.

The right size for this second sector is probably 1GiB.

Note that this will increase the number of groth params files (and corresponding verifying keys) which need to be downloaded. Only the (small) verifying keys *need* to be downloaded by those who don't want to seal the new sector sizes.

### Acceptance criteria

`go-filecoin` operators can choose a 1GiB (or other size as determined in the course of implementation) sector when setting up a node.

### Risks + pitfalls

Finding appropriate parameters in order to cater to different configurations is a hard problem. One benefit of multiple sector sizes is that we don't necessarily need to solve this perfectly. As larger sectors are added, lesser configurations will gradually be excluded. The risk is that we either jump too much and exclude too many too soon — or try to hard to tune *exactly* how the expansion proceeds.

In particular, this work will need to dovetail with adjustment to the security parameters (layers, challenges, etc.). The simplest way to avoid problems is probably to address this issue first, without any noise associated with varying other parameters yet. We can probably safely add a 1GiB sector without worrying about any particular developer configuration also.

### Where to begin

Find references to `TwoHundredFiftySixMiB` (or `SSB_TwoHundredFiftySixMiB`) and either extend to handle an equivalent `OneGiB` or change to wildcard (`_`) matching to handle all non-test cases the same.

**Comment by anorth** [comment]
With reference to additional Groth parameters, please ensure than a new or continuing go-filecoin developer's workflow doesn't download unnecessary files nor fail because parameters for 1GB sectors are not present. To most go-filecoin developers, sealing 1GB sectors is unnecessary for their work.

**Comment by laser** [comment]
@anorth 

> please ensure than a new or continuing go-filecoin developer's workflow doesn't download unnecessary files nor fail because parameters for 1GB sectors are not present. To most go-filecoin developers, sealing 1GB sectors is unnecessary for their work.

I think that `paramfetch` will need to be modified such that its consumers can specify the sizes of sectors for which they download Groth parameters. If a mining rig-operator (or go-filecoin developer) knows that they'll never use 1 GiB sectors (but that they _will_ use 1024 KiB sectors and 256 MiB sectors), they shouldn't have to download the 1 GiB sector's Groth parameters. I will write a story for the work required make this happen.

**Comment by laser** [comment]
Link to follow-up story: #629

**Comment by dignifiedquire** [comment]
@porcuquine are you actually working on this?

**Comment by dignifiedquire** [comment]
@laser how does this fold into your work of adding multiple sector sizes. I expect that we will need that to be finished before we can do this?

**Comment by porcuquine** [comment]
I'm waiting until multiple sector sizes are otherwise fully implemented. Adding a second sector size (and ensuring it works) could be considered the final piece of the multiple-sector-sizes puzzle.

@laser please let me know when you think we can absorb this change. I'm fine with someone else actually making the change at that time.

**Comment by laser** [comment]
@porcuquine 

> @laser please let me know when you think we can absorb this change. I'm fine with someone else actually making the change at that time.

The bare bones multiple sector sizes software is complete. 

There is some [question about whether or not adding new sector sizes will require changes to the protocol](https://github.com/filecoin-project/specs/pull/318), but that should not block the release of the 1GiB sector size.

Note that all miners will download all Groth parameters and verifying keys regardless of which sector sizes they will commit.

**Comment by anorth** [comment]
Please don't upgrade go-filecoin to use this until #629 is complete and miners do not download large parameter files that they don't need by default.
